ott/geometry/costs.py

```diff
@@ -94,6 +94,23 @@ def pairwise(self, x, y):
     return -2 * dot(x, y)
 
 
+@jax.tree_util.register_pytree_node_class
+class Cosine(CostFn):
+  """Cosine distance distance CostFn."""
+
+  def __init__(self, eps=1e-8):
+    super().__init__()
+    self._eps = eps
+
+  def pairwise(self, x, y):
+    eps = self._eps
+    x_norm = jnp.linalg.norm(x, axis=-1)
+    y_norm = jnp.linalg.norm(y, axis=-1)
+    cosine_similarity = dot(x, y) / jnp.maximum(x_norm * y_norm, eps)
```

Comment:
rather than maximum maybe use instead a +? this makes things a little bit smoother. i don't expect it to play a role, but...